common/lib/client/auth.js

```diff
@@ -97,7 +108,33 @@ var Auth = (function() {
 			Logger.logAction(Logger.LOG_ERROR, 'Auth()', msg);
 			throw new Error(msg);
 		}
-	}
+	};
+
+	/**
+	 * Update the authentication options for this client;
+	 * this will also update these options on the realtime connection
+	 * so that they take effect immediately
+	 * @param options
+	 */
+	Auth.prototype.setOptions = function(options) {
```

Comment:
`authorise()` ensures that credentials are available for the current options.

`setOptions()` changes the options.

If you wanted to set a new token with new capabilities, how would `authorise()` do that?

> I feel this is perhaps something we can live with until we support upgrade authentication in the protocol itself.

I'm not sure that the outcome would be much different in that case ...

> can you confirm what happens if the user is assigned less privileges than before, will that be rejected or will the new privileges be applied and if any channels are attached that are not supported, they will be detached. I assume there are cases where privileges may be downgraded and I think we should support probably this.

The `key` and `clientId` must be the same. Capabilities may be varied in either direction.